util/ranger/detacher.go

```diff
@@ -566,7 +566,9 @@ func ExtractEqAndInCondition(sctx sessionctx.Context, conditions []expression.Ex
 				// Maybe we can improve it later.
 				columnValues[i] = &valueInfo{mutable: true}
 			}
-			sctx.GetSessionVars().StmtCtx.MaybeOverOptimized4PlanCache = true
+			if expression.MaybeOverOptimized4PlanCache(sctx, conditions) {
+				return nil, conditions, nil, nil, false
```

Comment:
```
create table t (a int, b int, c int, d int, index idx_a_b(a, b, c));
prepare stmt from "select * from t where a = 2 and b = 3 and c >= ? and c <= ?";
set @v1 = 5, @v2 = 5;
execute stmt using @v1, @v2;
```
Here we return `nil` for `accesses`. Will the range `[2 3, 2 3]` be extracted after the change?